test/rpc_test.py

```diff
@@ -261,87 +265,82 @@ def test_duplicate_name(self):
 
     @dist_init(setup_model_parallel=False)
     def test_reinit(self):
-        dist.init_process_group(
-            backend=dist.Backend.GLOO,
-            init_method=self.init_method,
-            rank=self.rank,
-            world_size=self.world_size,
-        )
         rpc.init_model_parallel(
             self_name="worker{}".format(self.rank),
-            backend=TEST_CONFIG.rpc_backend,
+            backend=rpc.backend_registry.BackendType[TEST_CONFIG.rpc_backend_name],
             init_method=self.init_method,
             self_rank=self.rank,
             worker_name_to_id=self.worker_name_to_id,
         )
+
+        # This is for the below `dist.barrier`.
+        # For `RpcAgent` other than `ProcessGroupAgent`,
+        # no `_default_pg` is initialized.
+        if not dist.is_initialized():
+            dist.init_process_group(
+                backend="gloo",
+                init_method=self.init_method,
+                rank=self.rank,
+                world_size=self.world_size,
+            )
         # Wait for all init to complete.
         dist.barrier()
+
         with self.assertRaisesRegex(RuntimeError, "is already initialized"):
             rpc.init_model_parallel(
                 self_name="worker{}".format(self.rank),
-                backend=TEST_CONFIG.rpc_backend,
+                backend=rpc.backend_registry.BackendType[TEST_CONFIG.rpc_backend_name],
                 init_method=self.init_method,
                 self_rank=self.rank,
                 worker_name_to_id=self.worker_name_to_id,
             )
         rpc.join_rpc()
 
-    @dist_init(setup_model_parallel=False)
-    def test_init_invalid_backend(self):
-        with self.assertRaisesRegex(RuntimeError, "Unrecognized RPC backend"):
-            rpc.init_model_parallel(
-                self_name="worker{}".format(self.rank),
-                backend="invalid",
-                init_method=self.init_method,
-                self_rank=self.rank,
-                worker_name_to_id=self.worker_name_to_id,
-            )
-
     @dist_init(setup_model_parallel=False)
     def test_invalid_names(self):
-        dist.init_process_group(
-            backend=dist.Backend.GLOO,
-            init_method=self.init_method,
-            rank=self.rank,
-            world_size=self.world_size,
-        )
-
         with self.assertRaisesRegex(RuntimeError, "Worker name must match"):
             rpc.init_model_parallel(
                 self_name="abc*",
-                backend=TEST_CONFIG.rpc_backend,
+                backend=rpc.backend_registry.BackendType[TEST_CONFIG.rpc_backend_name],
                 init_method=self.init_method,
                 self_rank=self.rank,
                 worker_name_to_id=self.worker_name_to_id,
                 num_send_recv_threads=16,
             )
 
+        base_file_name = self.file_name
+        # Use a different file path for FileStore to avoid rendezvous mismatch.
+        self.file_name = base_file_name + "1"
```

Comment:
Will this file be removed from disk?

Reply:
None of the temp file is removed.

https://github.com/pytorch/pytorch/blob/6eef4690743a31ccc546f0c4f2d057d602e1e79e/test/common_distributed.py#L155